lm_eval/models/sae_steered_beta.py

```diff
@@ -0,0 +1,212 @@
+        12082, 240.0,gemma-scope-2b-pt-res-canonical,layer_20/width_16k/canonical, increase dogs
+        ...
+
+        Args:
+            csv_path: Path to the CSV file containing steering configurations
+            device: Device to place the model on
+
+        Returns:
+            InterventionModel with configured steering hooks
+        """
+        import pandas as pd
+        model = HookedSAETransformer.from_pretrained(base_name, device=device)
+        # Read steering configurations
+        df = pd.read_csv(csv_path)
+        # Create hooks for each row in the CSV
+        sae_cache = {}
+        hooks = []
+
+        def get_sae(sae_release, sae_id):
+            cache_key = (sae_release, sae_id)
+            if cache_key not in sae_cache:
+                sae_cache[cache_key] = SAE.from_pretrained(
+                    sae_release, sae_id, device=str(device)
+                )[0]
+            return sae_cache[cache_key]
+
+        for _, row in df.iterrows():
+            sae_release = row["sae_release"]
+            sae_id = row["sae_id"]
+            latent_idx = int(row["latent_idx"])
+            steering_coefficient = float(row["steering_coefficient"])
+            sae = get_sae(sae_release=sae_release, sae_id=sae_id)
+            sae.use_error_term = True
+            sae.eval()
+            model.add_sae(sae)
+            hook_action = row.get("hook_action", "add")
+            if hook_action == "add":
+                hook_name = f"{sae.cfg.hook_name}.hook_sae_input" # we aren't actually putting the input through the model
+                hook = partial(steering_hook_add_scaled_one_hot,
+                               sae=sae,
+                               latent_idx=latent_idx,
+                               steering_coefficient=steering_coefficient,
+                              )
+                model.add_hook(hook_name, hook)
+            elif hook_action == "clamp":
+                sae.add_hook("hook_sae_acts_post", partial(clamp_original, latent_idx=latent_idx, value=steering_coefficient))
+            elif hook_action == 'print':
+                sae.add_hook("hook_sae_acts_post", partial(print_sae_acts))
+            else:
+                raise ValueError(f"Unknown hook type: {hook_action}")
+            
+            
+
+        # Create and return the model
+        return cls(base_name=base_name, device=device, model=model)
+
+    def forward(self, *args, **kwargs):
+        # Handle both input_ids and direct tensor inputs
+        if "input_ids" in kwargs:
+            input_tensor = kwargs.pop("input_ids")  # Use pop to remove it
+        elif args:
+            input_tensor = args[0]
+            args = args[1:]  # Remove the first argument
+        else:
+            input_tensor = None
+        with torch.no_grad():  # I don't know why this no grad is necessary; I tried putting everything into eval mode. And yet, this is necessary to prevent CUDA out of memory exceptions.
```

Comment:
Eval mode doesn't set requires_grad to False, so you were getting OOMs because the autograd graph was being retained.